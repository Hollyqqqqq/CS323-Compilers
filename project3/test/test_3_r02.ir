FUNCTION main :
READ v2
READ v1
t5 := v2 / #100
t4 := t5 * #100
IF v2 == t4 GOTO label1
GOTO label2
LABEL label1 :
t11 := v2 / #400
t10 := t11 * #400
IF v2 == t10 GOTO label4
GOTO label5
LABEL label4 :
IF v1 == #2 GOTO label7
GOTO label8
LABEL label7 :
WRITE #29
GOTO label9
LABEL label8 :
t21 := v1 / #2
t20 := t21 * #2
IF v1 == t20 GOTO label10
GOTO label11
LABEL label10 :
WRITE #30
GOTO label9
LABEL label11 :
WRITE #31
LABEL label9 :
GOTO label6
LABEL label5 :
IF v1 == #2 GOTO label13
GOTO label14
LABEL label13 :
WRITE #28
GOTO label6
LABEL label14 :
t35 := v1 / #2
t34 := t35 * #2
IF v1 == t34 GOTO label16
GOTO label17
LABEL label16 :
WRITE #30
GOTO label6
LABEL label17 :
WRITE #31
LABEL label6 :
GOTO label3
LABEL label2 :
t45 := v2 / #4
t44 := t45 * #4
IF v2 == t44 GOTO label19
GOTO label20
LABEL label19 :
IF v1 == #2 GOTO label22
GOTO label23
LABEL label22 :
WRITE #29
GOTO label24
LABEL label23 :
t55 := v1 / #2
t54 := t55 * #2
IF v1 == t54 GOTO label25
GOTO label26
LABEL label25 :
WRITE #30
GOTO label24
LABEL label26 :
WRITE #31
LABEL label24 :
GOTO label3
LABEL label20 :
IF v1 == #2 GOTO label28
GOTO label29
LABEL label28 :
WRITE #28
GOTO label3
LABEL label29 :
t69 := v1 / #2
t68 := t69 * #2
IF v1 == t68 GOTO label31
GOTO label32
LABEL label31 :
WRITE #30
GOTO label3
LABEL label32 :
WRITE #31
LABEL label3 :
RETURN #0

