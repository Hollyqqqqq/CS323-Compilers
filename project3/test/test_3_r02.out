Program (1)
  IncludeList (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: main
        LP
        RP
      CompSt (2)
        LC
        DefList (3)
          Def (3)
            Specifier (3)
              TYPE: int
            DecList (3)
              Dec (3)
                VarDec (3)
                  ID: month
              COMMA
              DecList (3)
                Dec (3)
                  VarDec (3)
                    ID: year
            SEMI
          DefList (1)
        StmtList (4)
          Stmt (4)
            Exp (4)
              Exp (4)
                ID: year
              ASSIGN
              Exp (4)
                READ
                LP
                RP
            SEMI
          StmtList (5)
            Stmt (5)
              Exp (5)
                Exp (5)
                  ID: month
                ASSIGN
                Exp (5)
                  READ
                  LP
                  RP
              SEMI
            StmtList (6)
              Stmt (6)
                IF
                LP
                Exp (6)
                  Exp (6)
                    ID: year
                  EQ
                  Exp (6)
                    LP
                    Exp (6)
                      Exp (6)
                        Exp (6)
                          ID: year
                        DIV
                        Exp (6)
                          INT: 100
                      MUL
                      Exp (6)
                        INT: 100
                    RP
                RP
                Stmt (6)
                  CompSt (6)
                    LC
                    DefList (1)
                    StmtList (7)
                      Stmt (7)
                        IF
                        LP
                        Exp (7)
                          Exp (7)
                            ID: year
                          EQ
                          Exp (7)
                            LP
                            Exp (7)
                              Exp (7)
                                Exp (7)
                                  ID: year
                                DIV
                                Exp (7)
                                  INT: 400
                              MUL
                              Exp (7)
                                INT: 400
                            RP
                        RP
                        Stmt (7)
                          CompSt (7)
                            LC
                            DefList (1)
                            StmtList (8)
                              Stmt (8)
                                IF
                                LP
                                Exp (8)
                                  Exp (8)
                                    ID: month
                                  EQ
                                  Exp (8)
                                    INT: 2
                                RP
                                Stmt (8)
                                  CompSt (8)
                                    LC
                                    DefList (1)
                                    StmtList (9)
                                      Stmt (9)
                                        Exp (9)
                                          WRITE
                                          LP
                                          Exp (9)
                                            INT: 29
                                          RP
                                        SEMI
                                      StmtList (1)
                                    RC
                                ELSE
                                Stmt (11)
                                  IF
                                  LP
                                  Exp (11)
                                    Exp (11)
                                      ID: month
                                    EQ
                                    Exp (11)
                                      LP
                                      Exp (11)
                                        Exp (11)
                                          Exp (11)
                                            ID: month
                                          DIV
                                          Exp (11)
                                            INT: 2
                                        MUL
                                        Exp (11)
                                          INT: 2
                                      RP
                                  RP
                                  Stmt (11)
                                    CompSt (11)
                                      LC
                                      DefList (1)
                                      StmtList (12)
                                        Stmt (12)
                                          Exp (12)
                                            WRITE
                                            LP
                                            Exp (12)
                                              INT: 30
                                            RP
                                          SEMI
                                        StmtList (1)
                                      RC
                                  ELSE
                                  Stmt (15)
                                    Exp (15)
                                      WRITE
                                      LP
                                      Exp (15)
                                        INT: 31
                                      RP
                                    SEMI
                              StmtList (1)
                            RC
                        ELSE
                        Stmt (16)
                          CompSt (16)
                            LC
                            DefList (1)
                            StmtList (17)
                              Stmt (17)
                                IF
                                LP
                                Exp (17)
                                  Exp (17)
                                    ID: month
                                  EQ
                                  Exp (17)
                                    INT: 2
                                RP
                                Stmt (17)
                                  CompSt (17)
                                    LC
                                    DefList (1)
                                    StmtList (18)
                                      Stmt (18)
                                        Exp (18)
                                          WRITE
                                          LP
                                          Exp (18)
                                            INT: 28
                                          RP
                                        SEMI
                                      StmtList (1)
                                    RC
                                ELSE
                                Stmt (20)
                                  IF
                                  LP
                                  Exp (20)
                                    Exp (20)
                                      ID: month
                                    EQ
                                    Exp (20)
                                      LP
                                      Exp (20)
                                        Exp (20)
                                          Exp (20)
                                            ID: month
                                          DIV
                                          Exp (20)
                                            INT: 2
                                        MUL
                                        Exp (20)
                                          INT: 2
                                      RP
                                  RP
                                  Stmt (20)
                                    CompSt (20)
                                      LC
                                      DefList (1)
                                      StmtList (21)
                                        Stmt (21)
                                          Exp (21)
                                            WRITE
                                            LP
                                            Exp (21)
                                              INT: 30
                                            RP
                                          SEMI
                                        StmtList (1)
                                      RC
                                  ELSE
                                  Stmt (23)
                                    CompSt (23)
                                      LC
                                      DefList (1)
                                      StmtList (24)
                                        Stmt (24)
                                          Exp (24)
                                            WRITE
                                            LP
                                            Exp (24)
                                              INT: 31
                                            RP
                                          SEMI
                                        StmtList (1)
                                      RC
                              StmtList (1)
                            RC
                      StmtList (1)
                    RC
                ELSE
                Stmt (28)
                  IF
                  LP
                  Exp (28)
                    Exp (28)
                      ID: year
                    EQ
                    Exp (28)
                      LP
                      Exp (28)
                        Exp (28)
                          Exp (28)
                            ID: year
                          DIV
                          Exp (28)
                            INT: 4
                        MUL
                        Exp (28)
                          INT: 4
                      RP
                  RP
                  Stmt (28)
                    CompSt (28)
                      LC
                      DefList (1)
                      StmtList (29)
                        Stmt (29)
                          IF
                          LP
                          Exp (29)
                            Exp (29)
                              ID: month
                            EQ
                            Exp (29)
                              INT: 2
                          RP
                          Stmt (29)
                            CompSt (29)
                              LC
                              DefList (1)
                              StmtList (30)
                                Stmt (30)
                                  Exp (30)
                                    WRITE
                                    LP
                                    Exp (30)
                                      INT: 29
                                    RP
                                  SEMI
                                StmtList (1)
                              RC
                          ELSE
                          Stmt (32)
                            IF
                            LP
                            Exp (32)
                              Exp (32)
                                ID: month
                              EQ
                              Exp (32)
                                LP
                                Exp (32)
                                  Exp (32)
                                    Exp (32)
                                      ID: month
                                    DIV
                                    Exp (32)
                                      INT: 2
                                  MUL
                                  Exp (32)
                                    INT: 2
                                RP
                            RP
                            Stmt (32)
                              CompSt (32)
                                LC
                                DefList (1)
                                StmtList (33)
                                  Stmt (33)
                                    Exp (33)
                                      WRITE
                                      LP
                                      Exp (33)
                                        INT: 30
                                      RP
                                    SEMI
                                  StmtList (1)
                                RC
                            ELSE
                            Stmt (35)
                              CompSt (35)
                                LC
                                DefList (1)
                                StmtList (36)
                                  Stmt (36)
                                    Exp (36)
                                      WRITE
                                      LP
                                      Exp (36)
                                        INT: 31
                                      RP
                                    SEMI
                                  StmtList (1)
                                RC
                        StmtList (1)
                      RC
                  ELSE
                  Stmt (39)
                    CompSt (39)
                      LC
                      DefList (1)
                      StmtList (40)
                        Stmt (40)
                          IF
                          LP
                          Exp (40)
                            Exp (40)
                              ID: month
                            EQ
                            Exp (40)
                              INT: 2
                          RP
                          Stmt (40)
                            CompSt (40)
                              LC
                              DefList (1)
                              StmtList (41)
                                Stmt (41)
                                  Exp (41)
                                    WRITE
                                    LP
                                    Exp (41)
                                      INT: 28
                                    RP
                                  SEMI
                                StmtList (1)
                              RC
                          ELSE
                          Stmt (43)
                            IF
                            LP
                            Exp (43)
                              Exp (43)
                                ID: month
                              EQ
                              Exp (43)
                                LP
                                Exp (43)
                                  Exp (43)
                                    Exp (43)
                                      ID: month
                                    DIV
                                    Exp (43)
                                      INT: 2
                                  MUL
                                  Exp (43)
                                    INT: 2
                                RP
                            RP
                            Stmt (43)
                              CompSt (43)
                                LC
                                DefList (1)
                                StmtList (44)
                                  Stmt (44)
                                    Exp (44)
                                      WRITE
                                      LP
                                      Exp (44)
                                        INT: 30
                                      RP
                                    SEMI
                                  StmtList (1)
                                RC
                            ELSE
                            Stmt (46)
                              CompSt (46)
                                LC
                                DefList (1)
                                StmtList (47)
                                  Stmt (47)
                                    Exp (47)
                                      WRITE
                                      LP
                                      Exp (47)
                                        INT: 31
                                      RP
                                    SEMI
                                  StmtList (1)
                                RC
                        StmtList (1)
                      RC
              StmtList (50)
                Stmt (50)
                  RETURN
                  Exp (50)
                    INT: 0
                  SEMI
                StmtList (1)
        RC
    ExtDefList (1)
